### PR TITLE
add TestGameController

### PR DESCRIPTION
**Pull request by changlmasaki**
Add simple test for the GameController, some of them are from the previous codes.
We will add test for Tools before Saturday
Also we will help Jiaming to update the refactors on documentation

### CODE REVIEW

File: tests/kakuro/TestGameController.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+          assertEquals(conn.isClosed(), true);
+        } catch (SQLException e) {
+          // TODO Auto-generated catch block
+          e.printStackTrace();
+        }         
+    }
+
+    @Test
+    public void testSolveBoard() {
+    Boolean GUI = false;
```
Comment (binarytrails) [inline tests/kakuro/TestGameController.java]: indentation: 4 spaces in method cant be on same level as the method @changlmasaki

**Review by binarytrails — CHANGES_REQUESTED** [review]

File: tests/kakuro/TestGameController.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    GameController controller1 = new GameController(10,10, GUI);
+    GameModel model1 = new GameModel(10,10); 
+    model1.initBoard();  //initialize board
+    model1.board[0][1] = new BoardCell(BoardCell.CellType.FILLED10, 5);
+    model1.board[1][0] = new BoardCell(BoardCell.CellType.FILLED01,-1,4);
+    model1.board[1][1] = new BoardCell(BoardCell.CellType.INPUT,3,-1);
+    model1.board[2][1] = new BoardCell(BoardCell.CellType.INPUT,2,-1);
+    model1.board[1][2] = new BoardCell(BoardCell.CellType.INPUT,1,-1);
+    controller1.model = model1;
+    // Act
```
Comment (binarytrails) [inline tests/kakuro/TestGameController.java]: also shouldn't this be in three separates tests? the AAA part is one per each test

**Review by binarytrails — CHANGES_REQUESTED** [review]

File: tests/kakuro/TestGameController.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    model3.board[0][1] = new BoardCell(BoardCell.CellType.FILLED10, 5);
+    model3.board[1][0] = new BoardCell(BoardCell.CellType.FILLED11,-1,4);
+    model3.board[1][1] = new BoardCell(BoardCell.CellType.INPUT,2,-1);
+    model3.board[2][1] = new BoardCell(BoardCell.CellType.INPUT,3,-1);
+    model3.board[1][2] = new BoardCell(BoardCell.CellType.INPUT,2,-1);
+    controller3.model = model3;            
+    // Act
+    solved = controller3.solveBoard();
+    // Assert
+    assertEquals(solved, false);
```
Comment (binarytrails) [inline tests/kakuro/TestGameController.java]: ur doing like 3 different solves with 3 different boards in one test case

**Review by binarytrails — CHANGES_REQUESTED** [review]

**Comment by changlmasaki** [comment]
fix the space, and separate the 4 parts for testing the valid board, invalid board with wrong horizontal, vertical sum and invalid board with correct sum but duplicate inputs

File: tests/kakuro/TestGameController.java [hunk 4]
```diff
@@ -71,60 +72,109 @@ public void testDisconnectDatabase() {
...
+        controller1.model = model1;
+        // Act
+        solved = controller1.solveBoard();
+        // Assert
+        assertEquals(solved, true);
+    }
+    
+    @Test
+//  @brief Test Invalid Board with one wrong vertical sum
+    public void testSolveBoard2() {
```
Comment (binarytrails) [inline tests/kakuro/TestGameController.java]: @changlmasaki that is better!! but don't use a number but say what is does in the function name it is simplier for the documentation and comprehension like testBoardSolveInvalidBoardWithOneWongVerticalSum or something like that which follows a pattern easy to extend with Valid Invalid Many Good Horizontal etc

**Review by binarytrails — CHANGES_REQUESTED** [review]

**Comment by changlmasaki** [comment]
fix the test names

**Review by binarytrails — APPROVED** [review]

**Pull request merged**